contracts/utils/StrikeSelection.sol

```diff
@@ -5,13 +5,126 @@ pragma experimental ABIEncoderV2;
+    uint256 public delta;
+    // step in absolute terms at which we will increment
+    // (ex: 100 means we will move at increments of 100 points)
+    uint256 public step;
+
+    event DeltaSet(uint256 oldDelta, uint256 newDelta, address owner);
+    event StepSet(uint256 oldStep, uint256 newStep, address owner);
+
+    constructor(
+        address _optionsPremiumPricer,
+        uint256 _delta,
+        uint256 _step
+    ) {
+        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
+        require(_delta > 0, "!_delta");
+        require(_step > 0, "!_step");
+        optionsPremiumPricer = IOptionsPremiumPricer(_optionsPremiumPricer);
+        // ex: delta = 10
+        delta = _delta;
+        // ex: step = 1000
+        step = _step;
+    }
+
+    /**
+     * @notice Gets the strike price satisfying the delta value
+     * given the expiry timestamp and whether option is call or put
+     * @param expiryTimestamp is the unix timestamp of expiration
+     * @param isPut is whether option is put or call
+     */
+
+    function getStrikePrice(uint256 expiryTimestamp, bool isPut)
+        external
+        view
+        returns (uint256, uint256)
+    {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        // asset price
+        uint256 assetPrice = optionsPremiumPricer.getUnderlyingPrice();
+
+        // For each asset prices with step of 'margin' (down if put, up if call)
+        //   if asset's getOptionDelta(currStrikePrice, t) == (isPut ? 1 - delta:delta)
+        //   with certain margin of error
+        //        return strike price
+
+        bool pastDelta = false;
+        uint256 strike = assetPrice.sub(assetPrice % step);
+        uint256 targetDelta = isPut ? uint256(100).sub(delta) : delta;
+        uint256 prevDelta = 100;
+
+        while (true) {
+            uint256 currDelta =
+                optionsPremiumPricer.getOptionDelta(strike, expiryTimestamp);
+
+            //  If the current delta is between the previous
+            //  strike price delta and current strike price delta
+            //  then we are done
+            bool foundTargetStrikePrice =
+                isPut
+                    ? targetDelta >= prevDelta && targetDelta <= currDelta
+                    : targetDelta <= prevDelta && targetDelta >= currDelta;
+
+            if (foundTargetStrikePrice) {
+                uint256 upperBoundDiff =
+                    isPut
+                        ? sub(currDelta, targetDelta)
+                        : sub(prevDelta, targetDelta);
+                uint256 lowerBoundDiff =
+                    isPut
+                        ? sub(targetDelta, prevDelta)
+                        : sub(targetDelta, currDelta);
+                // for tie breaks (ex: 0.05 <= 0.1 <= 0.15) round to higher strike price for calls and lower strike price for puts
+                return (
```

Comment:
Before we return, we should do a `require(isPut ? strike < assetPrice : strike > assetPrice)` so we have a final sanity check